puzzle/cli.py

```diff
@@ -50,8 +50,10 @@ def cli(ctx, verbose, mode, variant_type, root):
     logger.debug('Booting up command line interface')
 
     if root is None:
-        root = os.path.join(os.environ['HOME'], '.puzzle')
-
+        try:
+            root = os.path.join(os.environ['HOME'], '.puzzle')
+        except:
```

Comment:
Could you except the error that is raised and log it, like:
```python
try:
    ...
except KeyError:
    logger.warning("Could not find $HOME")
    ...
```
or something like that?